tests/test_configuration_common.py

```diff
@@ -205,7 +207,131 @@ def check_config_arguments_init(self):
+        for attribute in attributes:
+            for modeling_source in modeling_sources:
+                # check if we can find `config.xxx`, `getattr(config, "xxx", ...)` or `getattr(self.config, "xxx", ...)`
+                if (
+                    f"config.{attribute}" in modeling_source
+                    or f'getattr(config, "{attribute}"' in modeling_source
+                    or f'getattr(self.config, "{attribute}"' in modeling_source
+                ):
+                    attribute_used = True
+                # Deal with multi-line cases
+                # TODO
+                elif (
+                    re.search(
+                        rf'getattr[ \t\v\n\r\f]*\([ \t\v\n\r\f]*(self\.)?config,[ \t\v\n\r\f]*"{attribute}"',
+                        modeling_source,
+                    )
+                    is not None
+                ):
+                    attribute_used = True
+                # `SequenceSummary` is called with `SequenceSummary(config)`
+                elif attribute in [
+                    "summary_type",
+                    "summary_use_proj",
+                    "summary_activation",
+                    "summary_last_dropout",
+                    "summary_proj_to_labels",
+                    "summary_first_dropout",
+                ]:
+                    if "SequenceSummary" in modeling_source:
+                        attribute_used = True
+                if attribute_used:
+                    break
+            if attribute_used:
+                break
+
+        # common and important attributes, although not always appear in the modeling files
+        attributes_to_allow = [
+            "bos_index",
+            "eos_index",
+            "pad_index",
+            "unk_index",
+            "mask_index",
+            "image_size",
+            "use_cache",
+        ]
+        attributes_used_in_generation = ["encoder_no_repeat_ngram_size"]
+
+        # Special cases to be allowed
+        case_allowed = True
+        if not attribute_used:
+            case_allowed = False
+            for attribute in attributes:
+
+                # Allow if the default value in the configuration class is different from the one in `PretrainedConfig`
+                if attribute in ["is_encoder_decoder"] and default_value is True:
+                    case_allowed = True
+                elif attribute in ["tie_word_embeddings"] and default_value is False:
+                    case_allowed = True
+
+                # Allow cases without checking the default value in the configuration class
+                elif attribute in attributes_to_allow + attributes_used_in_generation:
+                    case_allowed = True
+                elif attribute.endswith("_token_id"):
+                    case_allowed = True
+
+                # configuration class specific cases
+                if not case_allowed and check_fn is not None:
+                    case_allowed = check_fn(attribute)
```

Comment:
If none of the above **common** checks could not find the usage of an attribute or allow the edge case, we check further with the model specific rules given in `check_fn` that is passed from specific model testing files.